internal/cypress/v1/config.go

```diff
@@ -0,0 +1,487 @@
+	"github.com/saucelabs/saucectl/internal/msg"
+	"github.com/saucelabs/saucectl/internal/region"
+)
+
+// Config descriptors.
+var (
+	// Kind represents the type definition of this config.
+	Kind = "cypress"
+
+	// APIVersion represents the supported config version.
+	APIVersion = "v1"
+)
+
+// Project represents the cypress project configuration.
+type Project struct {
+	config.TypeDef `yaml:",inline" mapstructure:",squash"`
+	Defaults       config.Defaults        `yaml:"defaults" json:"defaults"`
+	DryRun         bool                   `yaml:"-" json:"-"`
+	ShowConsoleLog bool                   `yaml:"showConsoleLog" json:"-"`
+	ConfigFilePath string                 `yaml:"-" json:"-"`
+	CLIFlags       map[string]interface{} `yaml:"-" json:"-"`
+	Sauce          config.SauceConfig     `yaml:"sauce,omitempty" json:"sauce"`
+	Cypress        Cypress                `yaml:"cypress,omitempty" json:"cypress"`
+	// Suite is only used as a workaround to parse adhoc suites that are created via CLI args.
+	Suite         Suite                `yaml:"suite,omitempty" json:"-"`
+	Suites        []Suite              `yaml:"suites,omitempty" json:"suites"`
+	BeforeExec    []string             `yaml:"beforeExec,omitempty" json:"beforeExec"`
+	Docker        config.Docker        `yaml:"docker,omitempty" json:"docker"`
+	Npm           config.Npm           `yaml:"npm,omitempty" json:"npm"`
+	RootDir       string               `yaml:"rootDir,omitempty" json:"rootDir"`
+	RunnerVersion string               `yaml:"runnerVersion,omitempty" json:"runnerVersion"`
+	Artifacts     config.Artifacts     `yaml:"artifacts,omitempty" json:"artifacts"`
+	Reporters     config.Reporters     `yaml:"reporters,omitempty" json:"-"`
+	Env           map[string]string    `yaml:"env,omitempty" json:"env"`
+	Notifications config.Notifications `yaml:"notifications,omitempty" json:"-"`
+}
+
+// Suite represents the cypress test suite configuration.
+type Suite struct {
+	Name             string        `yaml:"name,omitempty" json:"name"`
+	Browser          string        `yaml:"browser,omitempty" json:"browser"`
+	BrowserVersion   string        `yaml:"browserVersion,omitempty" json:"browserVersion"`
+	PlatformName     string        `yaml:"platformName,omitempty" json:"platformName"`
+	Config           SuiteConfig   `yaml:"config,omitempty" json:"config"`
+	ScreenResolution string        `yaml:"screenResolution,omitempty" json:"screenResolution"`
+	Mode             string        `yaml:"mode,omitempty" json:"-"`
+	Timeout          time.Duration `yaml:"timeout,omitempty" json:"timeout"`
+	Shard            string        `yaml:"shard,omitempty" json:"-"`
+	Headless         bool          `yaml:"headless,omitempty" json:"headless"`
+	PreExec          []string      `yaml:"preExec,omitempty" json:"preExec"`
+	TimeZone         string        `yaml:"timeZone,omitempty" json:"timeZone"`
+	TestingType      string        `yaml:"testingType,omitempty" json:"testingType"`
```

Comment:
Framework specific settings for cypress are under `Config`, not directly in the `Suite` struct.